src/Microsoft.Extensions.ObjectPool/ObjectPool.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.ObjectPool
+{
+    public abstract class ObjectPool<T> where T : class
```

Comment:
Should we have `ObjectPool<T>.Shared` and `ObjectPoolProvider.Current` mechanisms?